lib/storage.js

```diff
@@ -0,0 +1,78 @@
+/*!
+ * Copyright (c) 2021 Digital Bazaar, Inc. All rights reserved.
+ */
+import database from 'bedrock-mongodb';
+import * as bedrock from 'bedrock';
+
+bedrock.events.on('bedrock-mongodb.ready', async () => {
+  await database.openCollections(['oauth2-client-accessToken']);
+
+  await database.createIndexes([{
+    // prevent duplicate tokens from being stored
+    collection: 'oauth2-client-accessToken',
+    fields: {'token.token': 1},
+    options: {unique: true, background: false}
+  }, {
+    // automatically expire tokens with an `expires` date field
+    collection: 'oauth2-client-accessToken',
+    fields: {'token.expires': 1},
+    options: {
+      partialFilterExpression: {'token.expires': {$exists: true}},
+      unique: false,
+      background: false,
+      expireAfterSeconds: 0
+    }
+  }]);
+});
+
+/**
+ * Queries the database for an existing access token for the client id.
+ *
+ * @param {object} options - Options to use.
+ * @param {string} options.client_id - The ID of the client.
+ *
+ * @returns {string} The access token if there is one, null if not.
+ */
+export async function get({client_id}) {
+  const query = {
+    'token.client_id': client_id
+  };
+  let token;
+  const record =
+    await database.collections['oauth2-client-accessToken'].findOne(query);
+  if(record && record.token) {
+    token = record.token;
+  }
+  return token;
+}
```

Comment:
We don't need this extra `token` var. We can just return the token if it is found. Also, this function does not return `null` right now when there is no token (it does not match the docs). It returns `undefined`. This suggestion updates that too -- however, it also just returns the full record. The storage API should not assume what people want to do with the records and it's cleaner to give them the whole record and it mirrors `insert()` (note: this also follows our standard practice for storage APIs). We very frequently find that we have to refactor to provide the whole record in the future for various reasons anyway.

Also, don't break after equals, break on parenthesis or on `.` separator -- or use a var to hold the collection name as that often looks clean too.

```suggestion
  const collection = database.collections['oauth2-client-accessToken'];
  const query = {'token.client_id': client_id};
  return collection.findOne(query);
}
```